storage-s3-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java

```diff
@@ -396,7 +401,12 @@ private void fixDeltaLog(FileSystem fs, ExternalCommitEntry entry) throws IOExce
                     fixDeltaLogPutCompleteDbEntry(entry);
                     LOG.info("fixed file {}", entry.fileName);
                     return;
-                } catch(Throwable e) {
+                } catch (java.nio.file.FileAlreadyExistsException e) {
+                    LOG.info("file already copied {}:", e.getClass().getSimpleName(), e);
+                    copied = true;
+                    // should we fixDeltaLogPutCompleteDbEntry after this to be safe?
```

Comment:
no, how it is now is fine.

since we catch this exception here, we do not do the `retry` count check below.

thus, we will go to the top of the method again, skip the copy since `copied=true`, and perform the `fixDeltaLogPutCompleteDbEntry`

Reply:
I'll remove that comment and leave as is